python-venv/cmstools/lib/k8s/k8s.py

```diff
@@ -45,19 +47,80 @@ def get_k8s_configmap_data(cm_name: str, cm_namespace: str = "default") -> JsonD
+    Get the current replica count for the specified deployment.
+    """
+    try:
+        deployment = apps_v1_api.read_namespaced_deployment(name=deployment_name, namespace=namespace)
+        return deployment.spec.replicas
+    except Exception as exc:
+        logger.exception("Error retrieving replica count for deployment '%s' in namespace '%s'",
+                         deployment_name, namespace)
+        raise CmstoolsException from exc
+
+def set_deployment_replicas(deployment_name: str, replicas: int, namespace: str = DEFAULT_NS) -> None:
+    """
+    Set the replica count for the specified deployment.
+    """
+    try:
+        body = {'spec': {'replicas': replicas}}
+        apps_v1_api.patch_namespaced_deployment_scale(name=deployment_name,
+                                                     namespace=namespace,
+                                                     body=body)
+    except Exception as exc:
+        logger.exception("Error scaling deployment '%s' in namespace '%s' to %d replicas",
+                         deployment_name, namespace, replicas)
+        raise CmstoolsException from exc
+
+def get_pod_count_for_deployment(deployment_name: str, namespace: str = DEFAULT_NS) -> int:
+    """
+    Return the number of pods in the specified namespace for a given deployment.
+    """
+    try:
+        # Get the deployment to extract its selector
+        deployment = apps_v1_api.read_namespaced_deployment(name=deployment_name, namespace=namespace)
+        selector = deployment.spec.selector.match_labels
+        label_selector = ",".join([f"{k}={v}" for k, v in selector.items()])
+        pods = k8s_client_api.list_namespaced_pod(namespace=namespace, label_selector=label_selector)
+        return len(pods.items)
+    except Exception as exc:
+        logger.exception("Error retrieving pod count for deployment '%s' in namespace '%s'", deployment_name, namespace)
+        raise CmstoolsException from exc
+
+def check_replicas_and_pods_scaled(deployment_name: str, expected_replicas: int, namespace: str = DEFAULT_NS) -> None:
+    """
+    Ensure deployment is scaled and all pods are terminated.
+    """
+    max_minutes = 5
+    max_time = time.time() + max_minutes*60
+    while True:
+        actual_replicas = get_deployment_replicas(deployment_name=deployment_name, namespace=namespace)
+
+        if actual_replicas == expected_replicas and get_pod_count_for_deployment(deployment_name=deployment_name) == 0:
+            logger.info("Deployment %s scaled to %d replicas and all pods terminated", deployment_name, expected_replicas)
+            return
+        if time.time() > max_time:
+            logger.error("Timeout: Deployment %s did not scale to %d replicas and terminate pods within %d minutes",
+                         deployment_name, expected_replicas, max_minutes)
+            raise CmstoolsException()
+
+        logger.info("Waiting for deployment %s to scale down and pods to terminate.", deployment_name)
```

Comment:
To be consistent with the log message format from the other functions in the module.

```suggestion
            logger.error(
                "Timeout: Deployment '%s' in namespace '%s' did not scale to %d replicas and terminate pods within %d minutes",
                deployment_name, namespace, expected_replicas, max_minutes
            )
            raise CmstoolsException()

        logger.info("Waiting for deployment '%s' in namespace '%s' to scale down and pods to terminate.",
                    deployment_name, namespace)
```